### PR TITLE
Renamings to make the code consistent with our terminology

### PR DESCRIPTION
**Pull request by aonyshchenko-strike**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix breaking (fix that would cause existing functionality to change)
- [x] Refactoring

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: tests/program_test.rs [hunk 1]
```diff
@@ -61,7 +62,7 @@ async fn init_program() {
         SlotId::new(0),
         AddressBookEntry {
             address: Pubkey::new_unique(),
-            name_hash: [0; 32],
+            name_hash: AddressBookEntryNameHash::new(&[0; 32]),
```
Comment (bholzman) [inline tests/program_test.rs:65]: nit: this could be `AddressBookEntryNameHash::zero()`

File: tests/program_test.rs [hunk 2]
```diff
@@ -269,20 +266,20 @@ async fn config_update_invalid_updates() {
 
     let address_book_entry = AddressBookEntry {
         address: Pubkey::new_unique(),
-        name_hash: [0; 32],
+        name_hash: AddressBookEntryNameHash::new(&[0; 32]),
```
Comment (bholzman) [inline tests/program_test.rs:269]: same for these

File: tests/program_test.rs [hunk 3]
```diff
@@ -861,19 +844,19 @@ async fn test_wallet_creation_not_signed_by_rent_collector() {
 }
 
 #[tokio::test]
-async fn test_wallet_creation_incorrect_hash() {
-    let mut context = utils::setup_wallet_tests(None).await;
+async fn test_balance_account_creation_incorrect_hash() {
+    let mut context = utils::setup_balance_account_tests(None).await;
+
+    let wrong_guid_hash = BalanceAccountGuidHash::new(&[0; 32]);
```
Comment (bholzman) [inline tests/program_test.rs:850]: could add a `::zero` for this struct and use it here too

File: tests/program_test.rs [hunk 4]
```diff
@@ -1706,9 +1682,10 @@ async fn test_transfer_sol_denied() {
 
 #[tokio::test]
 async fn test_transfer_wrong_destination_name_hash() {
-    let (mut context, balance_account) = utils::setup_wallet_tests_and_finalize(None).await;
+    let (mut context, balance_account) =
+        utils::setup_balance_account_tests_and_finalize(None).await;
 
-    context.destination_name_hash = [0; 32];
+    context.destination_name_hash = AddressBookEntryNameHash::new(&[0; 32]);
```
Comment (bholzman) [inline tests/program_test.rs:1688]: another one

File: tests/utils.rs [hunk 5]
```diff
@@ -149,20 +149,20 @@ pub async fn setup_program_config_update_test() -> ProgramConfigUpdateContext {
 
     let address_book_entry = AddressBookEntry {
         address: Pubkey::new_unique(),
-        name_hash: [0; 32],
+        name_hash: AddressBookEntryNameHash::new(&[0; 32]),
```
Comment (bholzman) [inline tests/utils.rs:152]: a couple more

**Review by bholzman — COMMENTED** [review]
Looks great! I'm going to merge my wrap/unwrap branch now, which will need to be adjusted for these changes too

**Comment by aonyshchenko-strike** [comment]
Rebased from master @bholzman please approve.

**Review by bholzman — APPROVED** [review]

**Pull request merged**